src/app/home-page/components/finish-modal/finish-modal.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from '@angular/core';
+import { TrackerDurationEnum } from 'src/app/shared-module/tracker/types/tracker-duration.enum';
+import { TrackerService } from '../../../shared-module/tracker/tracker.service';
+import { TrackerEventEnum } from '../../../shared-module/tracker/types/tracker-event.enum';
+
+@Component({
+  selector: 'app-finish-modal',
+  templateUrl: './finish-modal.component.html',
+  styleUrls: ['./finish-modal.component.scss'],
+})
+export class FinishModalComponent implements OnInit {
+  pomodoro_visibility = true;
+  break_visibility = true;
+
+  constructor(private tracker: TrackerService) {}
+
+  ngOnInit(): void {
+    this.tracker.event.subscribe((event) => {
+      if (event.eventType === TrackerEventEnum.finish)
+        {
+          if(event.duration === TrackerDurationEnum.pomodoro){
+            this.pomodoro_visibility = false;
+            setTimeout(() => (this.pomodoro_visibility = true), 7000);
+          } 
+          else if(event.duration === TrackerDurationEnum.longBreak)
+          {
+            this.break_visibility = false;
+            setTimeout(() => (this.break_visibility = true), 10000);
+          }
+          else if(event.duration === TrackerDurationEnum.shortBreak)
+          {
+            this.break_visibility = false;
+            setTimeout(() => (this.break_visibility = true), 10000);
+      }
```

Comment:
Since there are only 3 possible duration enums (pomodoro, longBreak, shortBreak) and the logic for handling the end of any of the breaks is the same, you can avoid duplication by rewriting your code as follows:

`else {`
`  this.break_visibility = false;`
`  setTimeout(() => (this.break_visibility = true), 10000);`
`}`